core-common/src/main/java/org/glassfish/jersey/internal/AbstractRuntimeDelegate.java

```diff
@@ -50,6 +52,8 @@
 
     private final Set<HeaderDelegateProvider> hps;
     private final Map<Class<?>, HeaderDelegate<?>> map;
+    private static final Object EPB_LOCK = new Object();
+    private static volatile EntityPartBuilderProvider cachedEntityPartBuilderProvider;
```

Comment:
You can get rid of both variables with an inner static class:
```
private static class Lazy {

    private static final EntityPartBuilderProvider cachedEntityPartBuilderProvider = findEntityPartBuilderProvider();

    /**
     * Obtain a {@code RuntimeDelegate} instance using the method described in {@link #getInstance}.
     *
     * @return an instance of {@code RuntimeDelegate}.
     */
    private static EntityPartBuilderProvider findEntityPartBuilderProvider() {
        for (final EntityPartBuilderProvider entityPartBuilder : ServiceFinder.find(EntityPartBuilderProvider.class)) {
            if (entityPartBuilder != null) {
                return entityPartBuilder;
            }
        }
        throw new IllegalArgumentException(LocalizationMessages.NO_ENTITYPART_BUILDER_FOUND());
    }
}
```

And later get the instance in this way. It will be loaded the first time is used.
`Lazy.cachedEntityPartBuilderProvider`

Reply:
We may utilize Jersey `LazyValue` for this